presto-spi/src/main/java/com/facebook/presto/spi/statistics/HistoricalPlanStatistics.java

```diff
@@ -16,34 +16,39 @@
 import com.facebook.drift.annotations.ThriftConstructor;
 import com.facebook.drift.annotations.ThriftField;
 import com.facebook.drift.annotations.ThriftStruct;
-import com.fasterxml.jackson.annotation.JsonProperty;
 
+import java.util.List;
 import java.util.Objects;
+import java.util.Optional;
 
 import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
 
 @ThriftStruct
 public class HistoricalPlanStatistics
 {
-    // TODO: Add more data for historical runs, like avg/min/max over last N days/runs
-    private final PlanStatistics lastRunStatistics;
+    private static final HistoricalPlanStatistics EMPTY = new HistoricalPlanStatistics(Optional.empty());
+    // Thrift values are stored as optional, to make them easier to change in future
+
+    // Output plan statistics from previous runs
```

Comment:
Actually, can you explain why making it optional makes it easier to change the field later?  Did you mean to set the thrift field type to optional?  Also, why won't an empty list do the trick? Would we have any different behavior on optional vs. empty list?

Reply:
Ah, I thought making it optional would automatically make it optional in thrift field type. I changed the actual thrift field requiredness setting, and removed optional from here. Thanks for the tip!